pipelines/tekton/gitops-update-pipeline/test-mlflow-retrieve-image-info-task.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: retrieve-image-info
+spec:
+  description: Retrieve location and SHA of the built image from the previous PipelineRun
+  params:
+  - name: namespace
+    type: string
+  - name: model-name
+    type: string
+  steps:
+  - name: get-image-sha
+    image: image-registry.openshift-image-registry.svc:5000/openshift/cli:latest
+    script: |
+      oc get -n $(params.namespace) pipelinerun --selector tekton.dev/pipeline=test-mlflow-image --sort-by=.status.completionTime -o jsonpath='{range .items[?(@.spec.params[0].name == "model-name")]}{.spec.params[?(@.name == "model-name")].value} {.status.results[?(@.name == "image-sha")].value} {.status.results[?(@.name == "target-registry-url")].value}{"\n"}{end}' | awk -v model=$(params.model-name) '$1 == model && NF == 3 { print $2, $3 }' | tail -1 | tee /dev/stderr | while read sha registry ; do echo -n "$sha" > $(results.image-sha.path) ; echo -n "$registry" > $(results.target-registry-url.path) ; done ;
```

Comment:
Since a Task is supposed to be a reusable self-contained action, I think this script should stay embedded in the `Pipeline` since it is hardcoded to `test-mlflow-image`.

Instead of querying a `PipelineRun`, can we simplify this workflow to rely on a known incluster ImageTag OR a ConfigMap that holds the current `ImagestreamTag` to reference for this Pipeline step

Reply:
The problem is, with a task defined merely as an entry in `tasks` in `Pipeline`, we are not able to have `results` there, IIRC. The only place I found `results` definition working is when I created it as a separate `Task`.

Yes, the task hardcodes `test-mlflow-image` -- but we have to hardcode something, to be able to reliably pick up the values from the previous PipelineRuns. I can turn it into a default of a parameter, if that's what you are suggesting.

The problem of using an ImageTag -- we want to be able to record in the pull request the SHA-256 of the exact image that was tested by the `test-mlflow-image` Pipeline. And the tag can change between the testing PipelineRun running the application and this PipelineRun finding the tag. That's why the whole logic of providing that value through `results` in the PipelineRuns.

Or do you suggest for the `test-mlflow-image` to set some very-well-known tag like `:testing-passed` any time the tests pass?

Reply:
Of course, my ideal approach would be to do it all in one Pipeline, test and file the pull request. But @piotrpdev was not fond of that.

Reply:
> Or do you suggest for the test-mlflow-image to set some very-well-known tag like :testing-passed any time the tests pass?

Your intent is sound but in the event we have to pass some config between PipelineRuns then we should use a configmap or annotation on the target object.

Reply:
I merged the changes after approval by @piotrpdev and after having https://github.com/opendatahub-io/ai-edge/issues/146 for the followup work. Let's consider this an interim quick-fix for the https://github.com/opendatahub-io/ai-edge/issues/125 problem and a baseline for further work.